mage/sh/lib.go

```diff
@@ -21,11 +22,16 @@ import (
 // to print output, if not, output will be hidden.
 // Stderr will work as normal
 func Run(cmd string, args ...string) error {
+	return RunWithEnv(nil, cmd, args...)
+}
+
+func RunWithEnv(env map[string]string, cmd string, args ...string) error {
```

Comment:
To my knowledge, we had no way to provide environment variables to any of these functions, but I needed this capability to set the `DOCKER_BUILDKIT` environment variable to instruct `docker build` to use buildkit. Not sure if adding a whole new set of functions that take an `env` map, in addition to existing functions, is the right thing to do. Maybe it's better to just refactor everything to always pass the env map.

Reply:
If we end up always using the RunWithEnv versions, we can clean things up in a follow-up ticket that removes the unused functions and possibly removes the WithEnv suffix.